src/containers/Bounty/SubmissionsAndCommentsCard.js

```diff
@@ -15,266 +15,309 @@ import { actions as bountyUIActions } from './reducer';
-
-      return (
-        <ListGroup.ListItem className={styles.submissionItem}>
-          <SubmissionItem
-            fulfillmentId={fulfillment_id}
-            fulfiller_name={name}
-            fulfiller_email={fulfiller_email}
-            fulfiller_address={fulfiller}
-            fulfiller_img={profile_image}
-            bounty={bounty}
-            url={url}
-            description={description}
-            dataHash={sourceDirectoryHash}
-            dataFileName={sourceFileName}
-            created={created}
-            accepted={accepted}
-            fulfiller_review={fulfiller_review || null}
-            issuer_review={issuer_review || null}
-            bountyBelongsToLoggedInUser={bountyBelongsToLoggedInUser}
-            submissionBelongsToLoggedInUser={submissionBelongsToLoggedInUser}
-            acceptFulfillment={() =>
-              initiateLoginProtection(() =>
-                initiateWalkthrough(() =>
-                  acceptFulfillment(bounty.id, fulfillment_id)
-                )
-              )
-            }
-            initiateLoginProtection={initiateLoginProtection}
-            showModal={showModal}
-            setRatingModal={setRatingModal}
-          />
-        </ListGroup.ListItem>
-      );
-    }, list);
-  };
+export const defaultTab = 'comments';
 
-  const renderComments = () => {
-    return map(comment => {
-      const { text, user, created } = comment;
-      const { name, profile_image, public_address } = user;
-
-      return (
-        <ListGroup.ListItem className={styles.commentItem}>
-          <CommentItem
-            name={name}
-            address={public_address}
-            img={profile_image}
-            text={text}
-            created={created}
-          />
-        </ListGroup.ListItem>
-      );
-    }, comments.list);
-  };
+export const mostInterestingTab = fulfillments => {
+  if (!fulfillments) {
+    return defaultTab;
+  }
 
-  let body = null;
-  let bodyClass = '';
+  if (fulfillments.list.length > 0) {
+    return 'submissions';
+  }
 
-  if (currentTab === 'submissions') {
-    body = <ListGroup>{renderFulfillments(fulfillments.list)}</ListGroup>;
+  return defaultTab;
+};
 
-    if (!fulfillments.list.length) {
-      bodyClass = styles.bodyLoading;
-      body = (
-        <div className={styles.zeroState}>
-          <ZeroState
-            title={'There are 0 submissions'}
-            text={'Submissions to this bounty will appear here.'}
-            iconColor="blue"
-            icon={['fal', 'level-up']}
-          />
-        </div>
-      );
+class SubmissionsAndCommentsCardComponent extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      tabLoaded: false
+    };
+  }
+
+  componentDidMount() {
+    const {
+      setActiveTabAction: setActiveTab,
+      fulfillments,
+      comments
+    } = this.props;
+
+    if (!this.state.tabLoaded && !fulfillments.loading && !comments.loading) {
+      this.setState({
```

Comment:
Not thrilled with this implementation, but after approaching this from different angles, this ended up being the cleanest way. We have to know if the logic for "most interesting tab" has been ran or not, and that should only happen when all the information is available and we're done loading. 

There's probably a few extra event loops that happen here that might be avoided if there's a better way, possibly at the `setActiveTab` level. 

As it stands, this works well and only adds the state for `tabLoaded`. Happy to improve this if you have suggestions!